src/mariadb_udf/config.c

```diff
@@ -0,0 +1,308 @@
+        }                              \
+        v = strchr(field, '=');        \
+        v++;                           \
+        tmp = strchr(v, ' ');          \
+        if (tmp == NULL) {             \
+            tmp = end;                 \
+        }                              \
+        val_len = tmp - v;             \
+    } while (0)
+
+int loadconfig(char* path, int* count, struct configuration** out)
+{
+    FILE* fp;
+    char host[256];
+    char port[256];
+    char* line = NULL, *end;
+    size_t len;
+    ssize_t read;
+    struct configuration* configurations = NULL;
+    ;
+    int conf_len = 0;
+    int rc = 0;
+
+    fp = fopen(path, "r");
+    if (fp == NULL) {
+        return errno;
+    }
+
+    while ((read = getline(&line, &len, fp)) != -1) {
+        char* value;
+        ssize_t val_len;
+        struct configuration conf;
+        memset(&conf, 0, sizeof(struct configuration));
+        if (line[0] == '#' || len < 3) {
+            continue;
+        }
+
+        // Chomp off the end
+        if (line[read - 1] == '\n') {
+            read--;
+        }
+
+        end = line + read;
+        get_val("type=", value, val_len);
+        // Check for a valid type
+        if (value == NULL || strncmp("syslog", value, val_len) != 0) {
+            continue;
+        }
+
+        rc = setup_syslog_header(&conf.syslog_header, line, read);
+        if (rc != 0) {
+            continue;
+        }
+
+        rc = setup_cef_header(&conf.cef_header, line, read);
+        if (rc != 0) {
+            continue;
+        }
+
+        get_val("host=", value, val_len);
+        if (value == NULL) {
+            cleanup_syslog_header(&conf.syslog_header);
+            cleanup_cef_header(&conf.cef_header);
+            continue;
+        }
+
+        strncpy(host, value, val_len);
+        host[val_len] = 0;
+
+        get_val("port=", value, val_len);
+        if (value == NULL) {
+            cleanup_syslog_header(&conf.syslog_header);
+            cleanup_cef_header(&conf.cef_header);
+            continue;
+        }
+
+        strncpy(port, value, val_len);
+        port[val_len] = 0;
+        rc = get_udp_socket(host, port, &conf.syslog_fd, (struct sockaddr_storage*)&conf.saddr, &conf.saddr_len);
+        if (rc != 0) {
+            cleanup_syslog_header(&conf.syslog_header);
+            cleanup_cef_header(&conf.cef_header);
+            continue;
+        }
+
+        get_val("namespaces=", value, val_len);
+        if (value == NULL) {
+            cleanup_syslog_header(&conf.syslog_header);
+            cleanup_cef_header(&conf.cef_header);
+            continue;
+        }
+
+        while (value < end) {
+            char* tmp = memchr(value, ',', end - value);
+            if (tmp == NULL) {
+                tmp = end;
+            }
+
+            val_len = tmp - value;
+            conf_len++;
+            configurations = realloc(configurations, sizeof(struct configuration) * conf_len);
```

Comment:
Ultimately, that memory is never released (on the last iteration of the loop). If it's fine, because it's executed only once by mysqld, at least mention it.